google/cloud/storage/client.cc

```diff
@@ -58,10 +58,10 @@ StatusOr<ObjectMetadata> Client::UploadFileSimple(
     std::string const& file_name, internal::InsertObjectMediaRequest request) {
   std::ifstream is(file_name);
   if (!is.is_open()) {
-    std::string msg = __func__;
-    msg += ": cannot open source file ";
-    msg += file_name;
-    google::cloud::internal::ThrowRuntimeError(msg);
+    std::ostringstream os;
+    os << __func__ << "(" << request << ", " << file_name
+       << "): cannot open upload file source";
```

Comment:
Switching to `ostringstream` can [potentially be inefficient](http://quick-bench.com/GxUDUArLNrqWFbo1S7WNUA3Li6s). Why not do `msg += request.contents()` instead?

Reply:
`request.contents()` does not have the same effect, I want to expose the parameters to the request, not just the media in the object.

And I am not too worried about performance on the error reporting path, errors are rare, if things are slow for errors that is hardly ever a problem.